src/schemes/boundary/open_boundary/system.jl

```diff
@@ -0,0 +1,477 @@
+                          reference_pressure=0.0,
+                          reference_density=first(boundary_zone.initial_condition.density))
+
+Open boundary system for in- and outflow particles.
+These open boundaries use the characteristic variables to propagate the appropriate values
+to the outlet or inlet and have been proposed by Lastiwka et al. (2009). For more information
+about the method see [description below](@ref method_of_characteristics).
+
+# Arguments
+- `boundary_zone`: Use [`InFlow`](@ref) for an inflow and [`OutFlow`](@ref) for an outflow boundary.
+
+# Keywords
+- `sound_speed`: Speed of sound.
+- `fluid_system`: The corresponding fluid system
+- `buffer_size`: Number of buffer particles.
+- `reference_velocity`: Reference velocity is either a function mapping each particle's coordinates
+                        and time to its velocity, an array where the ``i``-th column holds
+                        the velocity of particle ``i`` or, for a constant fluid velocity,
+                        a vector holding this velocity. Velocity is constant zero by default.
+- `reference_pressure`: Reference pressure is either a function mapping each particle's coordinates
+                        and time to its pressure, a vector holding the pressure of each particle,
+                        or a scalar for a constant pressure over all particles.
+                        Pressure is constant zero by default.
+- `reference_density`: Reference density is either a function mapping each particle's coordinates
+                       and time to its density, a vector holding the density of each particle,
+                       or a scalar for a constant density over all particles.
+                       Density is the density of the first particle in the initial condition by default.
+
+!!! warning "Experimental Implementation"
+	This is an experimental feature and may change in any future releases.
+"""
+struct OpenBoundarySPHSystem{BZ, NDIMS, ELTYPE <: Real, IC, FS, ARRAY1D, ARRAY2D, RV, RP,
+                             RD, B} <: System{NDIMS, IC}
+    initial_condition        :: IC
+    fluid_system             :: FS
+    mass                     :: ARRAY1D # Array{ELTYPE, 1}: [particle]
+    density                  :: ARRAY1D # Array{ELTYPE, 1}: [particle]
+    volume                   :: ARRAY1D # Array{ELTYPE, 1}: [particle]
+    pressure                 :: ARRAY1D # Array{ELTYPE, 1}: [particle]
+    characteristics          :: ARRAY2D # Array{ELTYPE, 2}: [characteristic, particle]
+    previous_characteristics :: ARRAY2D # Array{ELTYPE, 2}: [characteristic, particle]
+    sound_speed              :: ELTYPE
+    boundary_zone            :: BZ
+    flow_direction           :: SVector{NDIMS, ELTYPE}
+    reference_velocity       :: RV
+    reference_pressure       :: RP
+    reference_density        :: RD
+    buffer                   :: B
+    update_callback_used     :: Ref{Bool}
```

Comment:
This has to be reset to `false` at some point. Otherwise, one can run one proper simulation with a system and then reuse this system for simulations without `UpdateCallback`.
We might want to reset it in `semidiscretize`, and hope that no users solve the same `ODEProblem` again without an `UpdateCallback`.
To make this even safer, we could maybe also reset it in the initialization of the `InfoCallback`, arguing that this callback is almost always used.

Reply:
can't I just reset to `false` in `restart_with!`?

Reply:
No. You can create a system and then run a simulation. Then you can take the same system and run another simulation from the REPL.